musicround/__init__.py

```
@@ -177,148 +178,132 @@ def load_user(user_id):
-                    sp_oauth = SpotifyOAuth(
-                        client_id=app.config['SPOTIFY_CLIENT_ID'],
-                        client_secret=app.config['SPOTIFY_CLIENT_SECRET'],
-                        redirect_uri=url_for('users.spotify_callback', _external=True),
-                        scope=app.config['SPOTIFY_SCOPE']
-                    )
-                    
+    # Define token handling functions within create_app
+    def _app_fetch_token(name):
+        app.logger.debug(f"_app_fetch_token: Called for service '{name}', user: {current_user.id if current_user.is_authenticated else 'Unauthenticated'}")
+        if current_user.is_authenticated:
+            if name == 'spotify':
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)